### DISCUSSION TITLE
How to call a handler manually?

### DISCUSSION THREAD

**Discussion by moll-re** [python-telegram-bot/python-telegram-bot#2341] [Q&A]
I'd like to write a `/help` command that allows user to directly call another comand. As the commands are implemented through handlers, I'd like to manually trigger the dispatcher to enter a certain handler. 
I tried to do this through adding a job to the jobqueue,  but I'm not sure how to convert the handler to a callable.

Is there an intended way of doing what I want? My ultimate goal is to have an alias command that drops the user into a conversationhandler, at the stage he selected but right now I can't even get into the handler at all.

**Comment by Bibo-Joshi** [comment]
Sounds more like you'd want to add additional entry_points to your conversationhandler that take you directly to a specified state 🤔

**Reply by moll-re** [reply]
Ah yes, I hadn't thought about that. That sounds like a good idea. Thanks. Do you have any idea on how to launch a handler manually?

**Reply by Bibo-Joshi** [reply]
My reply reads as "There is no need to even try" ;) ofc you can still just call the callback associated with the handler.

**Reply by moll-re** [reply]
Thanks, but I have to ask, how do I access it?

**Reply by Bibo-Joshi** [reply]
`callback(update, context)`

**Reply by moll-re** [reply]
I'm sorry, could you please elaborate. 

If i have a created a conversationhandler called `h` I can access its entrypoints by `h.entry_points`. If I only set one entrypoint I can access its callable through `h.entry_points[0].callback`, which I can then call.

How would I do this to enter the whole conversationhandler (so that I can enter the subsequent states)?

**Reply by Bibo-Joshi** [reply]
If you know which callback you passed to which handler and which handler you assigned where in the conversation, you don't need to try and extract the callback from the handler via the conversationhandler - just access it directly.

> How would I do this to enter the whole conversationhandler (so that I can enter the subsequent states)?

As said above: Don't try that. Just add additional entry_points that do what you want.

**Reply by moll-re** [reply]
Ok, thanks. I guess I mixed up two questions here. I see what to do for an implementation of `alias` thanks. I'll reask my other question below for clarity.

**Comment by moll-re** [comment]
Independently I'm looking for a way to call a handler from within my code. (It would be a way to link two commands but being able to still call them individually through telegram). I'd therefore like to enter an existing handler from within my python-code. If I get this correctly the function `handle_update` of the `Handler` class would enable me to just do just that. I'm a bit lost when it comes to the parameter `check_result` (as specified in [https://python-telegram-bot.readthedocs.io/en/stable/telegram.ext.handler.html#telegram.ext.Handler.handle_update](https://python-telegram-bot.readthedocs.io/en/stable/telegram.ext.handler.html#telegram.ext.Handler.handle_update)). 

The documentation says it should be specified by `check_update` which defaults to `False`, so I specified `True` instead. I don't quite get the use of this function though.

Lastly, calling `handle_update` calls the entrypoint of my handler, but it seems to keep the state of the handler I just left. Is there a way to reset the state?

**Reply by Bibo-Joshi** [reply]
> Lastly, calling handle_update calls the entrypoint of my handler, but it seems to keep the state of the handler I just left. Is there a way to reset the state?

handlers don't have "states"

> I'm a bit lost when it comes to the parameter check_result

As per the docs, that's the return value of `handler.check_result(…)`


Still, I don't see a reason to try and call `handle_update` manually. You can just call the callback of the handler directly. The handlers are there to determine which updates should be handled by the callbacks. If for whatever reason you know that you want to have the callback of one specific handler executed for given `update` and `context`, why make the detour via the `handler` instead of just calling the `callback`?
